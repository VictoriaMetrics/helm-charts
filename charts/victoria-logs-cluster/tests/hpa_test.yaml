suite: hpa templates test
chart:
  version: 0.1.1
  appVersion: 0.1.0
templates:
  - hpa.yaml
tests:
  - it: hpa should match snapshot
    set:
      vmauth:
        enabled: true
        extraLabels:
          extraLabelName: authExtraLabelValue
        horizontalPodAutoscaler:
          enabled: true
      vminsert:
        extraLabels:
          extraLabelName: insertExtraLabelValue
        horizontalPodAutoscaler:
          enabled: true
      vmselect:
        extraLabels:
          extraLabelName: selectExtraLabelValue
        horizontalPodAutoscaler:
          enabled: true
      vmstorage:
        extraLabels:
          extraLabelName: storageExtraLabelValue
        horizontalPodAutoscaler:
          enabled: true
    asserts:
      - matchSnapshot: {}
  - it: hpa should match snapshot with fullnameOverride and extraLabels
    set:
      vminsert:
        name: insert
        fullnameOverride: vminsert-node
        extraLabels:
          extraLabelName: insertExtraLabelValue
        horizontalPodAutoscaler:
          enabled: true
      vmauth:
        enabled: true
        name: auth
        fullnameOverride: vmauth-node
        extraLabels:
          extraLabelName: authExtraLabelValue
        horizontalPodAutoscaler:
          enabled: true
      vmselect:
        name: select
        fullnameOverride: vmselect-node
        extraLabels:
          extraLabelName: selectExtraLabelValue
        horizontalPodAutoscaler:
          enabled: true
      vmstorage:
        name: storage
        fullnameOverride: vmstorage-node
        extraLabels:
          extraLabelName: storageExtraLabelValue
        horizontalPodAutoscaler:
          enabled: true
    asserts:
      - matchSnapshot: {}
