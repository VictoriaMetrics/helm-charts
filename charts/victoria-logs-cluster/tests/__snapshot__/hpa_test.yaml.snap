hpa should match snapshot:
  1: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app: vlinsert
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: 0.1.0-victorialogs
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: RELEASE-NAME-victoria-logs-cluster-vlinsert
      namespace: NAMESPACE
    spec:
      maxReplicas: 10
      metrics: []
      minReplicas: 2
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: RELEASE-NAME-victoria-logs-cluster-vlinsert
  2: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app: vmauth
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: v1.116.0
        extraLabelName: authExtraLabelValue
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: RELEASE-NAME-victoria-logs-cluster-vmauth
      namespace: NAMESPACE
    spec:
      maxReplicas: 10
      metrics: []
      minReplicas: 2
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: RELEASE-NAME-victoria-logs-cluster-vmauth
hpa should match snapshot with fullnameOverride and extraLabels:
  1: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app: vlinsert
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: 0.1.0-victorialogs
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: RELEASE-NAME-victoria-logs-cluster-vlinsert
      namespace: NAMESPACE
    spec:
      maxReplicas: 10
      metrics: []
      minReplicas: 2
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: RELEASE-NAME-victoria-logs-cluster-vlinsert
  2: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app: auth
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: v1.116.0
        extraLabelName: authExtraLabelValue
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: vmauth-node
      namespace: NAMESPACE
    spec:
      maxReplicas: 10
      metrics: []
      minReplicas: 2
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: vmauth-node
